File: URI/C/figurinhas.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
typedef unsigned int uint;
typedef unsigned long long int ulli;

typedef enum Bool {
    true = 1, false = 0
} bool;

int main(void){
    int casos;
    scanf("%d", &casos);
    for (int i = 0; i < casos; i++)
    {
        int a, b, mdc, resto, maior, menor;
        scanf("%d %d", &a, &b);
        maior = (a<b) ? a : b;
        menor = (a>b) ? a : b;

        resto = maior%menor;
        while (resto != 0)
        {
            maior = menor;
            menor = resto;
            resto = maior%menor;
        }
        mdc = menor;
        
        printf("%d\n", mdc);
    }
    return 0;
}

File: URI/Lua/figurinhas.lua
n = io.read("*number")
for i=1, n, 1
do
    a = io.read("*number")
    b = io.read("*number")
    if a>b then
        maior = a
        menor = b
    else
        maior = b
        menor = a
    end
    resto = maior%menor
    while(resto ~= 0)
    do
        maior = menor
        menor = resto
        resto = maior%menor
    end
    mdc = menor
    print(mdc)
end